addons/l10n_ch/models/res_bank.py

```diff
@@ -86,75 +86,106 @@ def find_number(self, s):
-                          self.acc_number,
-                          self.company_id.name,
-                          t_street_comp,
-                          number,
-                          self.company_id.zip,
-                          self.company_id.city,
-                          self.company_id.country_id.code,
-                          amount,
-                          currency,
-                          date_due,
-                          debitor.name,
-                          t_street_deb,
-                          number_deb,
-                          debitor.zip,
-                          debitor.city,
-                          debitor.country_id.code,
-                          ref_type,
-                          reference,
-                          communication)
-        qr_code_url = '/report/barcode/?type=%s&value=%s&width=%s&height=%s&humanreadable=1' % ('QR', werkzeug.url_quote_plus(qr_code_string), 256, 256)
-        return qr_code_url
-
-    def validate_swiss_code_arguments(self, currency, debitor):
-        currency = currency or self.company_id.currency_id
-        t_street_comp = '%s %s' % (self.company_id.street if (self.company_id.street != False) else '', self.company_id.street2 if (self.company_id.street2 != False) else '')
-        t_street_deb = '%s %s' % (debitor.street if (debitor.street != False) else '', debitor.street2 if (debitor.street2 != False) else '')
-        number = self.find_number(t_street_comp)
-        number_deb = self.find_number(t_street_deb)
-        if (t_street_comp == ' '):
-            t_street_comp = False
-        if (t_street_deb == ' '):
-            t_street_deb = False
-
-        if(currency.name == 'EUR'):
-            return (self.bank_id.l10n_ch_postal_eur and
-                    self.company_id.zip and
-                    self.company_id.city and
-                    self.company_id.country_id.code and
-                    (t_street_comp != False) and
-                    (t_street_deb != False) and
-                    debitor.zip and
-                    debitor.city and
-                    debitor.country_id.code and
-                    (number != False) and (number_deb != False))
-        elif(currency.name == 'CHF'):
-            return (self.bank_id.l10n_ch_postal_chf and
-                    self.company_id.zip and
-                    self.company_id.city and
-                    self.company_id.country_id.code and
-                    (t_street_comp != False) and
-                    (t_street_deb != False) and
-                    debitor.zip and
-                    debitor.city and
-                    debitor.country_id.code and
-                    (number != False) and (number_deb != False))
-        else:
-            return False
+    def build_swiss_code_url(self, amount, currency_name, not_used_anymore_1, debtor_partner, not_used_anymore_2, structured_communication, free_communication):
```

Comment:
Changing the method signature will break the report if the user don't do a -u. You should use a different method name instead.

Reply:
@nim-odoo Acutally, the parameters stay the same, I simply change their name to something more understandable (their content stays the same as before); so does it still count as a signature change? If so, why can it cause issues?

Reply:
changing the name of parameters is fine :)